Add Custom Hat Support

MrFawkes1337 opened the pull request:
This is a working Implementation of CorsacHats for TheOtherRoles. Custom Hats will be loaded from a folder named CustomHats in the Base Game Directory. You could probably clean it up a bit.

Eisbison commented:
Sorry for not looking at this for a while, kind of busy right now. I hope I'll be able to look at it tomorrow

MrFawkes1337 commented:
No rush. Im actually rewriting the Implementation, so might update this depending how it looks.

MrFawkes1337 commented:
Updated Implementation, All Hats are loaded from Resources/Hats instead for compatibility, and I included Corpse, Bretman, Pokimane, Toast, and Sykunno. I also added a reference to NotHunter's Modified Reactor, which can be found [Here](https://github.com/NotHunter101/Reactor) and doesn't use Modded Handshake so functions could be used without issues of compatibility.
